src/datepicker/datepicker.js

```diff
@@ -649,7 +663,7 @@ function($compile, $parse, $document, $rootScope, $position, dateFilter, datePar
           if (isNaN(date)) {
             return undefined;
           } else {
-            return date;
+            return timelessJsonMode ? dateFilter(date, 'yyyy-MM-dd') : date;
```

Comment:
This looks like it overrides the date format - shouldn't it be `dateFilter(date, dateFormat)`? What am I missing here?